src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs

```diff
@@ -170,19 +172,30 @@
      {:cell-padding-left "16px"
       :paginator :below
       :paginator-space 24
-      :resizable-columns? true})
+      :resizable-columns? true
+      :filterable? true})
    :get-initial-state
    (fn [{:keys [props]}]
      {:column-widths (mapv #(or (:starting-width %) 100) (:columns props))
-      :dragging? false})
+      :dragging? false
+      :filtered-data (:data props)})
    :render
-   (fn [{:keys [this state props]}]
+   (fn [{:keys [this state refs props]}]
      (let [paginator-above (= :above (:paginator props))
            paginator-below (= :below (:paginator props))
            paginator [Paginator {:ref "paginator"
-                                 :num-rows (count (:data props))
+                                 :num-rows (count (:filtered-data @state))
                                  :onChange #(react/call :handle-pagination-change this)}]]
        [:div {}
+        (when (:filterable? props)
+          (let [apply-filter #(swap! state assoc :filtered-data
+                               (react/call :filter-data this (trim (.-value (.getDOMNode (@refs "filter-field"))))))]
```

Comment:
`(react/call :filter-data this (-> (@refs "filter-field") .getDOMNode .-value trim))` would be another option if you like.